packages/jsii-reflect/README.md

```diff
@@ -0,0 +1,189 @@
+# jsii-reflect: strongly-type reflection library and tools for jsii
+
+**jsii-reflect** providers a reflection API for jsii type systems. It consumes
+`.jsii` manifest files (either directly or by traversing a closure of npm
+modules) and creates an in-memory object model that allows exploring the type
+system via a strongly-typed API.
+
+## Installation
+
+Install the npm module:
+
+```console
+$ npm i jsii-reflect
+```
+
+Import the `TypeSystem` class:
+
+```ts
+import { TypeSystem } from 'jsii-reflect';
+```
+
+## Loading Assemblies
+
+Create a `TypeSystem`:
+
+```ts
+const typesystem = new TypeSystem();
+```
+
+Now, we want to load assemblies into the type system via the `typesystem.load()`
+method.
+
+You can either pass in a path to a `.jsii` file, in which case you will have to
+eventually load the .jsii files of all the dependencies.
+
+```ts
+typesystem.load('jsii-calc/.jsii');
+typesystem.load('jsii-calc-lib/.jsii');
+typesystem.load('jsii-calc-base/.jsii');
+typesystem.load('jsii-calc-base-of-base/.jsii');
+```
+
+Or, you can point `load` to an npm module's directory, in which case the loader
+will transitively load all dependent modules:
+
+```ts
+// will automatically bring in all dependencies
+typesystem.load('jsii-calc');
```

Comment:
If I'm doing this and I want to do linting on the module I'm loading, is there a way for me to know the "primary" assembly?

Reply:
Good question. I'll add something.

Reply:
Added `filesystem.roots`